utils/update_dependencies.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Utility CLI to update dependency version requirements of the `setup.json`."""
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+
+import copy
+import os
+import click
+
+from validate_consistency import get_setup_json, write_setup_json
+
+FILENAME_SETUP_JSON = 'setup.json'
+SCRIPT_PATH = os.path.split(os.path.realpath(__file__))[0]
+ROOT_DIR = os.path.join(SCRIPT_PATH, os.pardir)
+FILEPATH_SETUP_JSON = os.path.join(ROOT_DIR, FILENAME_SETUP_JSON)
+DEFAULT_EXCLUDE_LIST = ['django', 'circus', 'numpy', 'pymatgen', 'ase', 'monty']
+
+
+@click.group()
+def cli():
+    """Utility to update dependency requirements for `aiida-core`.
+
+    Since `aiida-core` fixes the versions of almost all of its dependencies, once in a while these need to be updated.
+    This is a manual process, but this CLI attempts to simplify it somewhat. The idea is to remote all explicit version
+    restrictions from the `setup.json`, except for those packages where it is known that a upper limit is necessary.
+    This is accomplished by the command:
+
+        python update_dependencies.py unrestrict
+
+    The command will update the `setup.json` to remove all explicit limits, except for those packages specified by the
+    `--exclude` option. After this step, install `aiida-core` through pip with the `[all]` flag to install all optional
+    extra requirements as well. Since there are no explicit version requirements anymore, pip should install the latest
+    available version for each dependency.
+
+    Now run the unit test suite and reinstate dependency version requirements on packages that break the tests. Once all
+    the tests complete successfully, run the following command:
+
+        pip freeze > requirements.txt
```

Comment:
I guess all from here on down could be condensed to be a part of `python update_dependencies.py update`, and the `requirements` argument can be left out - unless you plan on having several different `requirements.txt` files that should be applied here?

Reply:
Yes, we could do all in a single command and perform a sys-call to `pip freeze` to get the versions, but since the whole thing is not fully automated anyway, I opted to keep it in two explicit steps. I feel this makes it clearer what is "happening".